### ISSUE TITLE
Error with tokens

### ISSUE THREAD

**Issue by theerfan** [plamere/spotipy#306]
Hi, I'm trying to manually add some albums to my saved ones, but I keep getting this error with a link that leads me to a page with some JSON code which says "no token provided", I think I have read the doc carefully enough, but I'm still unable to figure out the problem. 
[I have already done the first authorization thing where you have to give the program a link before starting everything.] 

```
import spotipy
import spotipy.util

from spotipy.oauth2 import SpotifyClientCredentials

username = //omitted
clientID = //omitted
access = 'user-library-modify'
clientSecret =  //omitted
redirectURL = 'http://localhost/'
token = spotipy.util.prompt_for_user_token(username =username, scope= access, client_id = clientID,
                                 client_secret = clientSecret, redirect_uri=redirectURL)
client_manager = SpotifyClientCredentials(clientID, clientSecret)
spotify = spotipy.Spotify(auth=token, client_credentials_manager= client_manager)

added_albums = //omitted #But a list of valid album ids that I sucessfully retrieved from spotify nevertheless

spotify.current_user_saved_albums_add(added_albums)
```

Thanks in advance.

**Comment by brobotic** [comment]
What version of spotipy are you using?

Having trouble reproducing this with your code. I created a new test app using your settings (same scope and redirect_uri) and successfully added an album to my saved music.

Still new myself, but the only thing that seems off to me is the line where you define your token. The documentation states that prompt_for_user_token's first argument is simply your username and not 'username=username'. Perhaps try changing that line to that? Like this:

`token = spotipy.util.prompt_for_user_token(username, scope=access, client_id=clientID, client_secret=clientSecret, redirect_uri=redirectURL)`

**Comment by theerfan** [comment]
Python 3.6.5 and Spotipy 2.4.4
Changed the line but I'm still getting the same link as before.

**Comment by brobotic** [comment]
Very odd indeed.

The VPN could be a factor, but I guess that would depend on the type of VPN and its configuration. spotipy is just opening https://accounts.spotify.com/authorize (along with some additional parameters in the URL) in your browser, and then their service will redirect you to whatever redirect_uri you set in your app configuration in the Developer portal on Spotify's website. Since you are using http://localhost/, I'm not terribly convinced that the VPN would interfere. With that said, hard to say if that's the issue.

Wish I could be of more help but I don't have any sort of VPN to test with / don't know which service you are using for that.

**Comment by stephanebruckert** [comment]
@theerfan you are currently using the authentication flows at the same time. To use the token and access your spotify profile data, you need to use the token while getting rid of the `client_credentials_manager`.

So change this:

```
client_manager = SpotifyClientCredentials(clientID, clientSecret)
spotify = spotipy.Spotify(auth=token, client_credentials_manager= client_manager) 
```

to:

    spotify = spotipy.Spotify(auth=token) 

Read about it here https://spotipy.readthedocs.io/en/2.6.1/#authorization-code-flow
